src/api/topup-virtual.api.ts

```diff
@@ -0,0 +1,28 @@
+import { InterceptedAPIHandler } from '../types/base.type';
+import { TopUpVirtualRequest } from '../types/topup-virtual.type';
+
+export const topupVirtual = ({ data, ws, session }: InterceptedAPIHandler) => {
+    const { topup_virtual, req_id } = data as TopUpVirtualRequest;
+    const virtual_account = session.accounts.find(
+        account => account.is_virtual === 1 && account.landing_company_name === 'virtual'
+    );
+
+    const virtual_balance = virtual_account?.balance;
+
+    const amount = 10000 - virtual_balance!;
+
+    const response = {
```

Comment:
response object best to put at the end if there are no other conditions modifying it